pkg/api/v1/clients/kube/resource_client.go

```diff
@@ -311,16 +248,23 @@ func (rc *ResourceClient) Delete(namespace, name string, opts clients.DeleteOpts
 
 	stats.Record(ctx, MInFlight.M(1))
 	defer stats.Record(ctx, MInFlight.M(-1))
-	if err := rc.kube.ResourcesV1().Resources(namespace).Delete(name, nil); err != nil {
+	if err := rc.crdClientset.ResourcesV1().Resources(namespace).Delete(name, nil); err != nil {
 		return errors.Wrapf(err, "deleting resource %v", name)
 	}
 	return nil
 }
 
 func (rc *ResourceClient) List(namespace string, opts clients.ListOpts) (resources.ResourceList, error) {
-	rc.sharedCache.startFactory(context.TODO(), rc.kubeClient)
 
-	lister, err := rc.sharedCache.sharedInformerFactory.GetLister(rc.crd.Type)
+	namespace = clients.DefaultNamespaceIfEmpty(namespace)
+	if err := rc.validateNamespace(namespace); err != nil {
+		return nil, err
+	}
+
+	// Will have no effect if the factory is already running
+	rc.sharedCache.Start(context.TODO())
```

Comment:
do we want to use `opts.WatchOpts.Ctx` here? @yuval-k

Reply:
In theory, as long as the credentials do not change the cache neednt expire.
Assuming opts.WatchOpts.Ctx expires on settings change which should be rare, i don't see a big different either way for performance. the question is of correctness:

is Start be called again when that context is canceled? what is the lifetime of sharedCache?

Reply:
That's a good point. Right now the controller (informers + workers) is stopped when the context that you pass to `Start(ctx)` is cancelled., which is never, since we are passing a todo context.

The most correct behavior would probably be to get the context in the resourceClient constructor and to propagate it to the controller. That way client code can determine their own lifecycle for the resourceClient.

Reply:
it's fine that they have never stopped, as long as it is consistent - 
i.e. when the next start is called it will re-use the existing object?

Reply:
After you call `start` the first time, subsequent invocations do not have any effect, and you can't register clients after the cache has been started. So there is no way to reuse it. You have to register all the clients that want to use the cache beforehand and then start it. After that it just lives forever. 

You're right that it's probably not a problem, given that AFAIK in our applications the client lifetime is the same as the one of the application, so there's no risk of leaking resources. Less than ideal, but good for now I guess.

Reply:
Maybe we could add a second constructor (not to break backwards-compatibility) in the form:
```go
NewKubeCache(done <-chan struct{})
```
That would be pretty quick.

Reply:
i would worry about backward-compatibility as there is no stable release.
lets talk about tomorrow morning - as I'm trying to understand why it doesn't work now vs before the change.

Reply:
Nothing really changed since the prior version, we already were passing a `ctx.TODO()` to the factory and never shutting it down (see [current `master`](https://github.com/solo-io/solo-kit/blob/bd125b289e404433f59b32148cb6eb54c488c5ec/pkg/api/v1/clients/kube/resource_client.go#L321)).